### PR TITLE
Calendar testing

### PR DESCRIPTION
**Pull request by SabrinaSnider**
Wrote the query() method for returning a list of available times for a group of people to meet. 

The method works by sorting the existing events and iterating through them. The "next available start time" for the meeting is tracked. If the meeting can fit between the "next available start time" and the start of the next event, that time is returned.

### CODE REVIEW

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java [hunk 1]
```diff
@@ -14,10 +14,54 @@
...
+        if (theoreticalStart + request.getDuration() <= event.getWhen().start()) {
+          // if theoretical end happens before the event starts, add it
+          availableTimes.add(TimeRange.fromStartEnd(theoreticalStart, event.getWhen().start(), false));
+        }
+
+        // move theoretical start pointer
+        theoreticalStart = event.getWhen().end();
+        continue;
+      } else if (theoreticalStart >= event.getWhen().start() && theoreticalStart < event.getWhen().end()) {
+        // if nextAvailableStart is in the middle of the event, move nextAvailableStart to after the event
```
Comment (brucezheng) [inline walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java:54]: What's nextAvailableStart?

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java [hunk 2]
```diff
@@ -14,10 +14,54 @@
...
+      if (theoreticalStart < event.getWhen().start()) {
+        if (theoreticalStart + request.getDuration() <= event.getWhen().start()) {
+          // if theoretical end happens before the event starts, add it
+          availableTimes.add(TimeRange.fromStartEnd(theoreticalStart, event.getWhen().start(), false));
+        }
+
+        // move theoretical start pointer
+        theoreticalStart = event.getWhen().end();
+        continue;
+      } else if (theoreticalStart >= event.getWhen().start() && theoreticalStart < event.getWhen().end()) {
```
Comment (brucezheng) [inline walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java:53]: Do you need to check for ```theoreticalStart >= event.getWhen().start()``` here? Isn't it true that ```!(theoreticalStart < event.getWhen().start())``` to make it to this block?

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java [hunk 3]
```diff
@@ -14,10 +14,54 @@
...
+    if (request.getAttendees().isEmpty()) return Arrays.asList(TimeRange.WHOLE_DAY);
+
+    Collection<TimeRange> availableTimes = new ArrayList<TimeRange>();
+
+    // if request is longer than a day, return no events
+    if (request.getDuration() > TimeRange.WHOLE_DAY.duration()) return availableTimes;
+
+    // make a sorted list of the events
+    List<Event> eventList = new ArrayList(events);
+    Collections.sort(eventList, (Event e1, Event e2) -> TimeRange.ORDER_BY_START.compare(e1.getWhen(), e2.getWhen()));
```
Comment (brucezheng) [inline walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java:35]: Are we able to assume that these events do not overlap?

Comment (SabrinaSnider) [reply walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java:35]: I don't think we can

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java (hunk 2 above)
Comment (SabrinaSnider) [reply walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java:53]: Oh right, good catch 👍

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java (hunk 1 above)
Comment (SabrinaSnider) [reply walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java:54]: Whoops, forgot to update the comment after changing variable names. It should read theoreticalStart.

**Pull request merged**